### PR TITLE
auth, etcdserver: forbid adding a user with empty name

### PR DESCRIPTION
**Pull request by mitake**
/cc @xiang90 @heyitsanthony

### CODE REVIEW

**Comment by xiang90** [comment]
lgtm. defer to @heyitsanthony

File: etcdserver/api/v3rpc/rpctypes/error.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ var (
 	ErrGRPCRootUserNotExist     = grpc.Errorf(codes.FailedPrecondition, "etcdserver: root user does not exist")
 	ErrGRPCRootRoleNotExist     = grpc.Errorf(codes.FailedPrecondition, "etcdserver: root user does not have root role")
 	ErrGRPCUserAlreadyExist     = grpc.Errorf(codes.FailedPrecondition, "etcdserver: user name already exists")
+	ErrGRPCUsernameEmpty        = grpc.Errorf(codes.InvalidArgument, "etcdserver: username is empty")
```
Comment (heyitsanthony) [inline etcdserver/api/v3rpc/rpctypes/error.go]: s/UsernameEmpty/UserEmpty and s/username/user name/ to be consistent with related error messages

**Comment by heyitsanthony** [comment]
lgtm following wording/naming consistency fixes

**Comment by mitake** [comment]
@heyitsanthony updated for the naming and wording, PTAL

**Comment by xiang90** [comment]
@mitake Thanks!

**Pull request merged**